modules/tests/shared/src/test/scala/CommandTest.scala

```diff
@@ -532,15 +532,19 @@ class CommandTest extends SkunkTest {
   }
 
   sessionTest("merge a record") { s =>
-    for {
+    (for {
       c <- s.prepare(insertCity).flatMap(_.execute(Garin))
       _ <- assert("completion",  c == Completion.Insert(1))
       c <- s.prepare(mergeCity).flatMap(_.execute(Garin.id))
       _ <- assert("merge", c == Completion.Merge(1))
       c <- s.prepare(selectCity).flatMap(_.option(Garin.id))
       _ <- assert("read", c == None)
+      _ <- s.execute(deleteCity)(Garin.id)
       _ <- s.assertHealthy
-    } yield "ok"
+    } yield "ok")
+      .recoverWith {
+        case SqlState.SyntaxError(ex) if ex.message.startsWith("""Syntax error at or near "MERGE"""") => s.execute(deleteCity)(Garin.id).as("ok")
+      }
```

Comment:
Hm, instead of this can we do something like `SELECT version();` to decide whether or not to run this test?

Reply:
Ah thanks! I was looking for querying direct support of merge, but that's much better of course.